test/metabase/driver/sql/query_processor/boolean_to_comparison_test.clj

```diff
@@ -0,0 +1,42 @@
+(ns metabase.driver.sql.query-processor.boolean-to-comparison-test
+  (:require
+   [clojure.test :refer :all]
+   [metabase.driver.sql.query-processor :as sql.qp]
+   [metabase.driver.sql.query-processor.boolean-to-comparison :as sql.qp.boolean-to-comparison]
+   [metabase.test :as mt]))
+
+(def ^:private true-value  [:value true  {:base_type :type/Boolean}])
+(def ^:private false-value [:value false {:base_type :type/Boolean}])
+
+(defn- basic-filter-query [expression]
+  (mt/mbql-query venues
+    {:expressions {"T" true-value, "F" false-value}
+     :fields      [[:expression "T"] [:expression "F"]]
+     :filter      expression}))
+
+(deftest ^:parallel boolean->comparison-test
+  (are [expression] (binding [sql.qp/*inner-query* (:query (basic-filter-query expression))]
+                      (= [:= expression true]
+                         (sql.qp.boolean-to-comparison/boolean->comparison expression)))
+    false
+    true
+    true-value
+    false-value
+    [:expression "T"]
+    [:expression "F"]
+    [:field "some-bool" {:base-type :type/Boolean}]
+    [:field 123 {:base-type :type/Boolean}]))
```

Comment:
Aren't negative tests (i.e., when no wrapping should happen) missing?

Reply:
Yes! I added them in the [other PR](https://github.com/metabase/metabase/pull/57723) that is stacked on top of this one because I was adding some additional tests there as well, but logically they belong to this PR.